File: tests/mt-test.cpp
#include "Cap2re.h"

#include <iostream>
using namespace std;

int main()
{
    cap2re::initialize();

    CameraManager *mgr;
    StringList images;

    mgr->detectCameras();

    cout << "Capture in single-threaded mode" << endl;
    images = mgr->captureFromAll("/tmp", false);

    cout << "Image files:" << endl;

    for(StringList::iterator img = images.begin(); img != images.end(); img++)
        cout << "   " << *img << endl;

    cout << endl;

    cout << "Capture in multi-threaded mode" << endl;
    images = mgr->captureFromAll("/tmp", true);

    cout << "Image files:" << endl;
    for(StringList::iterator img = images.begin(); img != images.end(); img++)
        cout << "   " << *img << endl;


    cap2re::shutdown();
    return 0;
}

File: tests/mt-test.py
import cap2re
from time import time

mgr = cap2re.Gphoto2CameraManager.instance()


mgr.detectCameras()

print "Testing shoot"
t1 = time()
mgr.captureFromAll("/tmp")
t2 = time() - t1
print "Time: ",t2
print

print "Single-thread shooting"
t1 = time()
mgr.captureFromAll("/tmp")
t2 = time() - t1
print "Time: ",t2
print

print "Multi-threaded shooting"
t1 = time()
mgr.captureFromAll("/tmp", True)
t2 = time() - t1
print "Time: ",t2
